### PR TITLE
Fix broken tests

### PR DESCRIPTION
**Pull request by grafnu**

### CODE REVIEW

File: testing/python_lib/build_config.py [hunk 1]
```diff
@@ -246,7 +265,8 @@ def main(argv):
     else:
         raise Exception('Unkown topology type: %s' % topo_type)
 
-    config_map = proto_dict(faucet_config)
+    config_map = cleanup_config(proto_dict(faucet_config))
```
Comment (henry54809) [inline testing/python_lib/build_config.py:268]: Can the fix be in proto_dict instead? Or for some situations, it’s necessary to have str keys?

File: testing/python_lib/build_config.py [hunk 2]
```diff
@@ -201,6 +202,24 @@ def create_corp_faucet_config(self):
         return FaucetConfig(dps=dps, version=2)
 
 
+def cleanup_keys(map_in):
+    """proto_dict converstaion converts int keys to strings, which causes problems with faucet."""
```
Comment (henry54809) [inline testing/python_lib/build_config.py:206]: Is “converstaion” conversion?

File: testing/python_lib/build_config.py (hunk 1 above)
Comment (grafnu) [reply testing/python_lib/build_config.py:268]: There are definitely cases where str keys are valid, so I can't think of any easy way to make it more general without being overly complicated (as in, dynamically investigating the proto to see what it's supposed to be and then patching it up).

File: testing/python_lib/build_config.py (hunk 2 above)
Comment (grafnu) [reply testing/python_lib/build_config.py]: updated

File: testing/python_lib/build_config.py [hunk 3]
```diff
@@ -201,6 +202,24 @@ def create_corp_faucet_config(self):
...
+    return map_out
+
+
+def cleanup_config(config_map):
+    """cleanup the necessary dict entries"""
+    for dp_id in config_map['dps']:
+        bad_map = config_map['dps'][dp_id]['interfaces']
+        config_map['dps'][dp_id]['interfaces'] = cleanup_keys(bad_map)
+    if 'vlans' in config_map:
+        config_map['vlans'] = cleanup_keys(config_map['vlans'])
```
Comment (henry54809) [inline testing/python_lib/build_config.py:219]: It's really not necessary to have another function for this, you can replace cleanup_keys with this line `dict(iter((int(k), v) for k, v in config_map['vlans'].items()))`

**Review by henry54809 — APPROVED** [review]

**Comment by grafnu** [comment]
I think using the dict/iter approach is good (better than explicit for
loop), but I think the function itself is still worthwhile since it
provides a named construct for it, and prevent replicating the code itself
(which would end up in multiple places...)...

**Pull request merged**